### PR TITLE
refactoring and modularizing code to separate the functions into thei…

### PR DESCRIPTION
**Pull request by modernNeo**
…r respective classes

### CODE REVIEW

**Comment by TitanVJ** [comment]
in the whois command change the part thats `"```" + memberString + "```"` to be `"```\n" + memberString + "```"`(line 102). this'll solve the n-1 bug that would occur on merging.

**Comment by modernNeo** [comment]
@TitanVJ if you see anything that needs correcting, you can do
`git pull`
`git checkout modularize_and_refactor`
`git checkout -b new_branch_name`
//work on the fix
`git commit -m "commit message"`
`git push origin new_branch_name`
and then do a PR from new_branch_name to modularize_and_refactor

**Comment by TitanVJ** [comment]
in that case ill just work on my own branch with embeds and new sfu commands, since the embeds take care of the bug anyway

**Comment by TitanVJ** [comment]
pls run code before approving and testing every command. as of rn it doesnt work.

**Comment by TitanVJ** [comment]
2018-09-04 15:58:06 = WARNING - 
2018-09-04 15:58:41 = ERROR - [main.py on_command_error()] Ignoring exception in command newrole:
2018-09-04 15:58:41 = WARNING - Traceback (most recent call last):

2018-09-04 15:58:41 = WARNING - 
2018-09-04 15:58:41 = WARNING -   File "/home/vj/.local/lib/python3.6/site-packages/discord/ext/commands/core.py", line 62, in wrapped
    ret = yield from coro(*args, **kwargs)

2018-09-04 15:58:41 = WARNING - 
2018-09-04 15:58:41 = WARNING -   File "/mnt/c/Users/VJ/Documents/SFU/csss/wall_e/commands_to_load/RoleCommands.py", line 13, in newrole
    logger.info("[RoleCommands newrole()] "+ctx.message.author+" called newrole with following argument: roleToAdd="+roleToAdd)

2018-09-04 15:58:41 = WARNING - 
2018-09-04 15:58:41 = WARNING - TypeError: must be str, not Member

2018-09-04 15:58:41 = WARNING - 
2018-09-04 15:58:41 = WARNING - 
The above exception was the direct cause of the following exception:


2018-09-04 15:58:41 = WARNING - 
2018-09-04 15:58:41 = WARNING - Traceback (most recent call last):

2018-09-04 15:58:41 = WARNING - 
2018-09-04 15:58:41 = WARNING -   File "/home/vj/.local/lib/python3.6/site-packages/discord/ext/commands/bot.py", line 886, in invoke
    yield from ctx.command.invoke(ctx)

2018-09-04 15:58:41 = WARNING - 
2018-09-04 15:58:41 = WARNING -   File "/home/vj/.local/lib/python3.6/site-packages/discord/ext/commands/core.py", line 498, in invoke
    yield from injected(*ctx.args, **ctx.kwargs)

2018-09-04 15:58:41 = WARNING - 
2018-09-04 15:58:41 = WARNING -   File "/home/vj/.local/lib/python3.6/site-packages/discord/ext/commands/core.py", line 71, in wrapped
    raise CommandInvokeError(e) from e

2018-09-04 15:58:41 = WARNING - 
2018-09-04 15:58:41 = WARNING - discord.ext.commands.errors.CommandInvokeError: Command raised an exception: TypeError: must be str, not Member

**Comment by modernNeo** [comment]
`2018-09-04 15:58:06 = WARNING -
2018-09-04 15:58:41 = ERROR - [main.py on_command_error()] Ignoring exception in command newrole:
2018-09-04 15:58:41 = WARNING - Traceback (most recent call last):

2018-09-04 15:58:41 = WARNING -
2018-09-04 15:58:41 = WARNING - File "/home/vj/.local/lib/python3.6/site-packages/discord/ext/commands/core.py", line 62, in wrapped
ret = yield from coro(*args, **kwargs)

2018-09-04 15:58:41 = WARNING -
2018-09-04 15:58:41 = WARNING - File "/mnt/c/Users/VJ/Documents/SFU/csss/wall_e/commands_to_load/RoleCommands.py", line 13, in newrole
logger.info("[RoleCommands newrole()] "+ctx.message.author+" called newrole with following argument: roleToAdd="+roleToAdd)

2018-09-04 15:58:41 = WARNING -
2018-09-04 15:58:41 = WARNING - TypeError: must be str, not Member

2018-09-04 15:58:41 = WARNING -
2018-09-04 15:58:41 = WARNING -
The above exception was the direct cause of the following exception:

2018-09-04 15:58:41 = WARNING -
2018-09-04 15:58:41 = WARNING - Traceback (most recent call last):

2018-09-04 15:58:41 = WARNING -
2018-09-04 15:58:41 = WARNING - File "/home/vj/.local/lib/python3.6/site-packages/discord/ext/commands/bot.py", line 886, in invoke
yield from ctx.command.invoke(ctx)

2018-09-04 15:58:41 = WARNING -
2018-09-04 15:58:41 = WARNING - File "/home/vj/.local/lib/python3.6/site-packages/discord/ext/commands/core.py", line 498, in invoke
yield from injected(*ctx.args, **ctx.kwargs)

2018-09-04 15:58:41 = WARNING -
2018-09-04 15:58:41 = WARNING - File "/home/vj/.local/lib/python3.6/site-packages/discord/ext/commands/core.py", line 71, in wrapped
raise CommandInvokeError(e) from e

2018-09-04 15:58:41 = WARNING -
2018-09-04 15:58:41 = WARNING - discord.ext.commands.errors.CommandInvokeError: Command raised an exception: TypeError: must be str, not Member`

**Comment by modernNeo** [comment]
> 2018-09-04 15:58:06 = WARNING -
2018-09-04 15:58:41 = ERROR - [main.py on_command_error()] Ignoring exception in command newrole:
2018-09-04 15:58:41 = WARNING - Traceback (most recent call last):

2018-09-04 15:58:41 = WARNING -
2018-09-04 15:58:41 = WARNING - File "/home/vj/.local/lib/python3.6/site-packages/discord/ext/commands/core.py", line 62, in wrapped
ret = yield from coro(*args, **kwargs)

2018-09-04 15:58:41 = WARNING -
2018-09-04 15:58:41 = WARNING - File "/mnt/c/Users/VJ/Documents/SFU/csss/wall_e/commands_to_load/RoleCommands.py", line 13, in newrole
logger.info("[RoleCommands newrole()] "+ctx.message.author+" called newrole with following argument: roleToAdd="+roleToAdd)

2018-09-04 15:58:41 = WARNING -
2018-09-04 15:58:41 = WARNING - TypeError: must be str, not Member

2018-09-04 15:58:41 = WARNING -
2018-09-04 15:58:41 = WARNING -
The above exception was the direct cause of the following exception:

2018-09-04 15:58:41 = WARNING -
2018-09-04 15:58:41 = WARNING - Traceback (most recent call last):

2018-09-04 15:58:41 = WARNING -
2018-09-04 15:58:41 = WARNING - File "/home/vj/.local/lib/python3.6/site-packages/discord/ext/commands/bot.py", line 886, in invoke
yield from ctx.command.invoke(ctx)

2018-09-04 15:58:41 = WARNING -
2018-09-04 15:58:41 = WARNING - File "/home/vj/.local/lib/python3.6/site-packages/discord/ext/commands/core.py", line 498, in invoke
yield from injected(*ctx.args, **ctx.kwargs)

2018-09-04 15:58:41 = WARNING -
2018-09-04 15:58:41 = WARNING - File "/home/vj/.local/lib/python3.6/site-packages/discord/ext/commands/core.py", line 71, in wrapped
raise CommandInvokeError(e) from e

2018-09-04 15:58:41 = WARNING -
2018-09-04 15:58:41 = WARNING - discord.ext.commands.errors.CommandInvokeError: Command raised an exception: TypeError: must be str, not Member

**Comment by wnfldchen** [comment]
I will be working on this to turn it into an environment variable, such that it can be tested properly both locally and automated-ly

**Comment by modernNeo** [comment]
@wnfldchen would you like the merging to wait till before or after you turn some things into env variables?

**Comment by modernNeo** [comment]
@TitanVJ please do further tests with updated code

**Comment by wnfldchen** [comment]
Let's have it wait i think

**Comment by modernNeo** [comment]
@wnfldchen if you want to close the PR while working out the kinks, go ahead

**Comment by TitanVJ** [comment]
btw the typecasting fix commit u added @modernNeo only fixed the issue in one line. its in every line that uses logger in the RoleCommands.py and Administration.py

**Comment by wnfldchen** [comment]
its on the test server now, test away to your hearts content on it

File: commands_to_load/Administration.py [hunk 1]
```diff
@@ -9,7 +9,7 @@
 class Administration():
 
 	async def botManager(self, ctx):
-		for managers in self.admin['BOT_MANAGERS']:
+		for managers in self.admin_list['BOT_MANAGERS']:
```
Comment (modernNeo) [inline commands_to_load/Administration.py]: why rename it?

Comment (modernNeo) [reply commands_to_load/Administration.py]: @wnfldchen

Comment (wnfldchen) [reply commands_to_load/Administration.py]: it was broken before, it was "red underlined" error unresolved reference

Comment (wnfldchen) [reply commands_to_load/Administration.py]: (`self.admin` doesnt exist, `self.admin_list` does)

Comment (modernNeo) [reply commands_to_load/Administration.py]: huh, i missed that. thanks for alerting me to it

**Comment by modernNeo** [comment]
will work on remindme function tomorrow night

**Comment by modernNeo** [comment]
@TitanVJ and @wnfldchen go head and try to break it if possible please

**Comment by wnfldchen** [comment]
logging on the test server is done and it seems functional on the test server.

**Comment by modernNeo** [comment]
@TitanVJ and @wnfldchen are you both confident that this now works?

File: main.py [hunk 2]
```diff
@@ -1,207 +1,140 @@
...
-                ctx = await bot.get_context(msg)
-                if ctx.valid and testenv.TestCog.check_test_environment(ctx):
-                    fmt = '<@{0}> ```{1}```'
-                    await ctx.send(fmt.format(ctx.message.author.id, ctx.message.content))
-            except Exception as error:
-                print('Ignoring exception when generating reminder:', file=sys.stderr)
-                traceback.print_exception(type(error), error, error.__traceback__, file=sys.stderr)
-        await asyncio.sleep(2)
-
+    global BOT_LOG_CHANNEL
```
Comment (modernNeo) [inline main.py]: @wnfldchen what does putting `global` here do?

Comment (modernNeo) [reply main.py]: then I will remove `global` from it :)

**Comment by wnfldchen** [comment]
logs should be persistent now

**Pull request merged**